packages/grid/_modules_/grid/models/props/DataGridProps.ts

```diff
@@ -1,54 +1,343 @@
+import { GridClasses } from '../../gridClasses';
+import { GridCellParams, GridRowParams } from '../params';
+import { GridFilterModel } from '../gridFilterModel';
+import { GridInputSelectionModel, GridSelectionModel } from '../gridSelectionModel';
+import { GridInitialState } from '../gridState';
+import { GridSlotsComponentsProps } from '../gridSlotsComponentsProps';
+
+/**
+ * The props users can give to the `DataGrid` component.
+ */
+export type DataGridProps = Omit<
+  Partial<DataGridSimpleOptions> & DataGridComplexOptionsBeforeProcessing & DataGridOtherProps,
+  DataGridForcedPropsKey
+> & {
+  pagination?: true;
+};
+
+/**
+ * The props of the `DataGrid` component after the pre-processing phase.
+ */
+export interface DataGridProcessedProps
+  extends DataGridSimpleOptions,
+    DataGridComplexOptionsAfterProcessing,
+    DataGridOtherProps {}
+
+/**
+ * The props of the `DataGrid` component after the pre-processing phase that the user should not be able to override.
+ * Those are usually used in feature-hook for which the pro-plan has more advanced features (eg: multi-sorting, multi-filtering, ...).
+ */
+export type DataGridForcedPropsKey =
+  | 'apiRef'
+  | 'checkboxSelectionVisibleOnly'
+  | 'disableMultipleColumnsFiltering'
+  | 'disableMultipleColumnsSorting'
+  | 'disableMultipleSelection'
+  | 'disableColumnReorder'
+  | 'disableColumnResize'
+  | 'throttleRowsMs'
+  | 'hideFooterRowCount'
+  | 'pagination'
+  | 'signature';
+
+/**
+ * The `DataGrid` options with a default value that must be merged with the value given through props.
+ */
+export interface DataGridComplexOptionsAfterProcessing {
+  components: GridSlotsComponent;
+  localeText: GridLocaleText;
+}
+
+/**
+ * The `DataGrid` options with a default value that must be merged with the value given through props.
+ */
+export interface DataGridComplexOptionsBeforeProcessing {
+  /**
+   * Overrideable components.
+   */
+  components?: Partial<GridSlotsComponent>;
+  /**
+   * Set the locale text of the grid.
+   * You can find all the translation keys supported in [the source](https://github.com/mui-org/material-ui-x/blob/HEAD/packages/grid/_modules_/grid/constants/localeTextConstants.ts) in the GitHub repository.
+   */
+  localeText?: Partial<GridLocaleText>;
+}
 
 /**
- * The grid component react props before applying the default values.
+ * The `DataGrid` options with a default value overridable through props
+ * None of the entry of this interface should be optional, they all have default values and `DataGridProps` already applies a `Partial<DataGridSimpleOptions>` for the public interface
+ * The controlled model do not have a default value at the prop processing level, so they must be defined in `DataGridOtherProps`
+ * TODO: add multiSortKey
  */
-export interface GridInputComponentProps
-  extends Partial<GridSimpleOptions>,
-    Partial<GridMergedOptions>,
-    GridComponentOtherProps {}
+export interface DataGridSimpleOptions {
```

Comment:
I never know the different between `DataGridSimpleOptions` and `DataGridOtherProps`. Since you're changing them I propose the new names.

```diff
-DataGridSimpleOptions
-DataGridOtherProps
+DataGridPropsWithDefaultValue
+DataGridPropsWithoutDefaultValue
```

Reply:
Renamed :+1: